src/game/level.h

```diff
@@ -14,17 +15,53 @@
 #include <string>
 #include <vector>
 
+struct Background {
+  i64 x;
+  i64 y;
+  std::vector<u32> background_data;
+};
+
+struct Objects {
+  i64 x;
+  i64 y;
+  u32 width;
+  u32 height;
+  u32 tileset_id;
+  std::string script;
+};
+
+struct Collisions {
+  i64 x;
+  i64 y;
+  std::vector<u64> collisions_data;
+};
+
 class Level {
 private:
   lua_State *lua;
+  EntityManager *ent_manager;
   simdjson::ondemand::parser json_parser;
-  simdjson::ondemand::document json_data;
+  simdjson::ondemand::document json_data, id_json_data;
+
+  std::vector<Background> background_content;
```

Comment:
you probably should rename these variables to simply `background`, `objects`, `collisions` without the `_content` suffix